server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/AuthenticationContract.scala

```diff
@@ -19,133 +19,214 @@
 import io.restassured.authentication.NoAuthScheme
-import io.restassured.http.{Header, Headers}
+import io.restassured.http.Header
 import org.apache.http.HttpStatus.{SC_OK, SC_UNAUTHORIZED}
 import org.apache.james.GuiceJamesServer
-import org.apache.james.jmap.rfc8621.contract.AuthenticationContract._
-import org.apache.james.jmap.rfc8621.contract.Fixture._
+import org.apache.james.jmap.rfc8621.contract.Fixture.{ACCEPT_RFC8621_VERSION_HEADER, ALICE, ALICE_PASSWORD, AUTHORIZATION_HEADER, BOB, BOB_BASIC_AUTH_HEADER, BOB_PASSWORD, DOMAIN, DOMAIN_WITH_SPACE, ECHO_REQUEST_OBJECT, INVALID_JWT_TOKEN, UNKNOWN_USER_TOKEN, USER_TOKEN, getHeadersWith, toBase64, _}
 import org.apache.james.jmap.rfc8621.contract.tags.CategoryTags
 import org.apache.james.utils.DataProbeImpl
-import org.junit.jupiter.api.{BeforeEach, Tag, Test}
+import org.junit.jupiter.api.{BeforeEach, Nested, Tag, Test}
 
-object AuthenticationContract {
-  private val AUTHORIZATION_HEADER: String = "Authorization"
-}
 
 trait AuthenticationContract {
   @BeforeEach
-  def setUp(server: GuiceJamesServer): Unit = {
+  def setup(server: GuiceJamesServer): Unit = {
     server.getProbe(classOf[DataProbeImpl])
-      .fluent()
-      .addDomain(DOMAIN.asString())
-      .addUser(BOB.asString(), BOB_PASSWORD)
-      .addDomain(_2_DOT_DOMAIN.asString())
-      .addUser(ALICE.asString(), ALICE_PASSWORD)
+      .fluent
+      .addDomain(DOMAIN.asString)
+      .addDomain(_2_DOT_DOMAIN.asString)
+      .addUser(USER.asString, USER_PASSWORD)
+      .addUser(ALICE.asString, ALICE_PASSWORD)
+      .addUser(BOB.asString, BOB_PASSWORD)
 
     requestSpecification = baseRequestSpecBuilder(server)
-      .setAuth(new NoAuthScheme())
+        .setAuth(new NoAuthScheme)
       .build
   }
 
-  @Test
-  def postShouldRespondUnauthorizedWhenNoAuthorizationHeader(): Unit = {
-    `given`()
-      .header(ACCEPT.toString, ACCEPT_RFC8621_VERSION_HEADER)
-      .body(ECHO_REQUEST_OBJECT)
-    .when()
-      .post()
-    .then
-      .statusCode(SC_UNAUTHORIZED)
-  }
+  @Nested
+  class BothAuthenticationMechanisms {
+    @Tag(CategoryTags.BASIC_FEATURE)
+    @Test
+    def shouldRespond200WhenBasicAuthValidAndJWTInvalid(): Unit = {
+      `given`
+        .headers(getHeadersWith(BOB_BASIC_AUTH_HEADER))
+        .header(new Header(AUTHORIZATION_HEADER, s"Bearer $UNKNOWN_USER_TOKEN"))
+        .body(ECHO_REQUEST_OBJECT)
+      .when
+        .post
+      .`then`
+        .statusCode(SC_OK)
+    }
 
-  @Test
-  @Tag(CategoryTags.BASIC_FEATURE)
-  def postShouldRespond200WhenHasCredentials(): Unit = {
-    val authHeader: Header = new Header(AUTHORIZATION_HEADER, s"Basic ${toBase64(s"${BOB.asString}:$BOB_PASSWORD")}")
-    `given`()
-      .headers(getHeadersWith(authHeader))
-      .body(ECHO_REQUEST_OBJECT)
-    .when()
-      .post()
-    .then
-      .statusCode(SC_OK)
-  }
+    @Tag(CategoryTags.BASIC_FEATURE)
+    @Test
+    def shouldRespond200WhenJWTAuthValidAndBasicAuthInvalid(): Unit = {
```

Comment:
I'm a bit surprised about that, why do we should to allow wrong credentials to be present in a request?

Reply:
We are missing these tests on `draft`, but currently we find the first valid authentication method.

I believe if we wish to change that behaviour we should open another ticket, and fix it on draft too.

Reply:
> I'm a bit surprised about that, why do we should to allow wrong credentials to be present in a request?

Currently, we are accepting both kind of Authentication?

Reply:
> I believe if we wish to change that behaviour we should open another ticket, and fix it on draft too.
BTW, i agree with Benoit, we should change it in another PR.

Reply:
I don't mind you handle that in another PR.

So, in my opinion we should reject a request as soon as a set of credentials is wrong AND we should not allow more than one set of credentials (because otherwise we would have to handle cases where it's actually two different identity)